pkg/controller/replicaset/replica_set_test.go

```diff
@@ -650,16 +653,41 @@ func TestUpdatePods(t *testing.T) {
 	testRSSpec2.Name = "barfoo"
 	informers.Extensions().V1beta1().ReplicaSets().Informer().GetIndexer().Add(&testRSSpec2)
 
-	// case 1: We put in the podLister a pod with labels matching testRSSpec1,
-	// then update its labels to match testRSSpec2.  We expect to receive a sync
-	// request for both replica sets.
+	isController := true
+	controllerRef1 := metav1.OwnerReference{UID: testRSSpec1.UID, APIVersion: "v1", Kind: "ReplicaSet", Name: testRSSpec1.Name, Controller: &isController}
+	controllerRef2 := metav1.OwnerReference{UID: testRSSpec2.UID, APIVersion: "v1", Kind: "ReplicaSet", Name: testRSSpec2.Name, Controller: &isController}
+
+	// case 1: Pod with a ControllerRef
 	pod1 := newPodList(informers.Core().V1().Pods().Informer().GetIndexer(), 1, v1.PodRunning, labelMap1, testRSSpec1, "pod").Items[0]
+	pod1.OwnerReferences = []metav1.OwnerReference{controllerRef1}
 	pod1.ResourceVersion = "1"
 	pod2 := pod1
 	pod2.Labels = labelMap2
 	pod2.ResourceVersion = "2"
 	manager.updatePod(&pod1, &pod2)
-	expected := sets.NewString(testRSSpec1.Name, testRSSpec2.Name)
+	expected := sets.NewString(testRSSpec1.Name)
+	for _, name := range expected.List() {
+		t.Logf("Expecting update for %+v", name)
+		select {
+		case got := <-received:
+			if !expected.Has(got) {
+				t.Errorf("Expected keys %#v got %v", expected, got)
+			}
+		case <-time.After(wait.ForeverTestTimeout):
+			t.Errorf("Expected update notifications for replica sets within 100ms each")
```

Comment:
Actually I don't know. I copied this from other tests in this file. Maybe the text of the error is outdated. I will make it less specific.